langchain/chat_models/base.py

```diff
@@ -246,6 +248,7 @@ def _generate_with_cache(
         messages: List[BaseMessage],
         stop: Optional[List[str]] = None,
         run_manager: Optional[CallbackManagerForLLMRun] = None,
+        prompt_index: Optional[int] = None,
```

Comment:
This is unfortunately not backwards compatible... will need to do an additional `inspect.signature`. Is there a better way to achieve that?